From 4eaecec81538361d00f1588a6ae0b83cf30e7ed3 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Mon, 13 Feb 2012 11:40:18 +0800
Subject: [PATCH 275/505] ENGR00174300 vpu: Fix the QBUF failed with rotation
 on i.MX6 in unit test

VIDOC_QBUF failed when rotation or tiled2linear feature is enabled on i.MX6.
The patch fixed the issue.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 test/mxc_vpu_test/dec.c      | 1 +
 test/mxc_vpu_test/display.c  | 5 ++++-
 test/mxc_vpu_test/vpu_test.h | 1 +
 3 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/test/mxc_vpu_test/dec.c b/test/mxc_vpu_test/dec.c
index b1d064a..9825ac3 100644
--- a/test/mxc_vpu_test/dec.c
+++ b/test/mxc_vpu_test/dec.c
@@ -1437,6 +1437,7 @@ decoder_allocate_framebuffer(struct decode *dec)
 		dec->rot_buf_count = ((dec->cmdl->dst_scheme == PATH_V4L2) ||
 				(dec->cmdl->dst_scheme == PATH_IPU)) ? 2 : 1;
 		dec->extrafb += dec->rot_buf_count;
+		dec->post_processing = 1;
 	}
 
 	/*
diff --git a/test/mxc_vpu_test/display.c b/test/mxc_vpu_test/display.c
index c904a20..6f18742 100644
--- a/test/mxc_vpu_test/display.c
+++ b/test/mxc_vpu_test/display.c
@@ -1014,7 +1014,10 @@ int v4l_put_data(struct decode *dec, int index, int field, int fps)
 
 	disp->ncount++;
 
-	threshold = dec->regfbcount - dec->minfbcount;
+	if (dec->post_processing)
+		threshold = dec->rot_buf_count - 1;
+	else
+		threshold = dec->regfbcount - dec->minfbcount;
 	if (disp->queued_count > threshold) {
 		if (vpu_v4l_performance_test) {
 			sem_post(&disp->avaiable_dequeue_frame);
diff --git a/test/mxc_vpu_test/vpu_test.h b/test/mxc_vpu_test/vpu_test.h
index 57db05e..482dcd8 100644
--- a/test/mxc_vpu_test/vpu_test.h
+++ b/test/mxc_vpu_test/vpu_test.h
@@ -216,6 +216,7 @@ struct decode {
 	Rect picCropRect;
 	int reorderEnable;
 	int tiled2LinearEnable;
+	int post_processing;
 
 	DecReportInfo mbInfo;
 	DecReportInfo mvInfo;
-- 
1.8.0

